File: csharp/DataStructures/CustomStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp
{
    public class CustomStack
    {
        private int lastElement;
        private string[] stackArray;
        private int stackSize;

        public CustomStack(int size) {
            stackSize = size;
            stackArray = new string[stackSize];
            lastElement = 0;
        }

        public void Push(string element) {
            if (lastElement + 1 == stackSize) {
                // double array size
            }
            stackArray[lastElement++] = element;
        }

        public string Pop() {
            string element = "empty";

            if (lastElement > -1) {
                element = stackArray[lastElement--];
            }

            return element;
        }

        public string Peak() {
            string element = "empty";

            if (lastElement > -1) {
                element = stackArray[lastElement];
            }

            return element;
        }

        private 
    }
}

File: java/src/com/codyferguson/CustomStack.java
package com.codyferguson;

public class CustomStack {
    private static final int STANDARD_SIZE = 10;

    private int[] stackArray;
    private int size;
    private int top;

    public CustomStack() {
        size = STANDARD_SIZE;
        stackArray = new int[size];
        top = 0;
    }

    public CustomStack(int size) {
        this.size = size;
        stackArray = new int[size];
        top = 0;
    }

    public int pop() {
        return -1;
    }

    public void push(int element) {

    }

    public
}
